### ISSUE TITLE
nimble-dialog includes extra header padding when subtitle is excluded

### ISSUE THREAD

**Issue by kroeschl** [ni/nimble#1588]
# 🐛 Bug Report

`nimble-dialog` has a lot of spacing below the header when the subtitle is excluded.

## 💻 Repro or Code Sample

```html
<nimble-dialog footer-hidden>
   <span slot="title">Some title</span>
   <p>Some content</p>
</nimble-dialog>
```
Comparison of with and without subtitle:
![image](https://github.com/ni/nimble/assets/60117794/78636d92-9d53-4d34-816d-5d24b296a989)

## 🤔 Expected Behavior

I would expect the subtitle padding to not be visible at all of it's not populated.

## 😯 Current Behavior

We end up with a rather large blank area for the subtitle.

## 💁 Possible Solution

I have no idea. This may be intentional, in which case we'll just close out this issue.

## 🔦 Context

I have a dialog without a subtitle that ends up looking a little wonky:
![image](https://github.com/ni/nimble/assets/60117794/b48520b3-06d1-4b4e-b46b-549fd23ac56e)

## 🌍 Your Environment

* OS & Device: Windows 10
* Browser: Chrome 117.0.5938.132
* Version: @ni/nimble-angular 16.8.2

**Comment by mollykreis** [comment]
The header styles came from visual design, so I'll let @NIbokeefe confirm that reserving space for a subtitle in the dialog is still what is desired.

If you don't want the footer on the bottom of the dialog, you can add the `footer-hidden` attribute to the component. That will adjust the bottom padding of the dialog.

**Comment by kroeschl** [comment]
Oops, I missed the `footer-hidden` attribute somehow. 😬 I'll update the issue to scope it down to the subtitle.

**Comment by NIbokeefe** [comment]
@kroeschl @mollykreis Sorry for the delayed response. 

This was by design and not a development error.  We looked into both scenarios of removing the space or keeping the space for the subtitle.  We concluded on keeping the space which improved consistency of where the content always lands.  This mainly optimized for a series of dialogs where it may or may not have a subtitle one frame to the next.  We were okay with the negative space and made a judgement call at the time, that the pros outweighed the cons.  Possible assumptions may have been that the subtitle would be used more commonly.

Side note: The spacing consideration for a warning (shown in this mockup)
is outdated because we now have an inline warning banner that replaces the floating text.

<img width="1234" alt="Screen Shot 2023-10-19 at 12 44 56 PM" src="https://github.com/ni/nimble/assets/106594349/220f965f-1bc9-4f71-a3cf-8ac82440cd5f">
